control/controlclient/auto.go

```diff
@@ -694,9 +694,18 @@ func (c *Auto) StartLogout() {
 func (c *Auto) Logout(ctx context.Context) error {
 	c.logf("client.Logout()")
 
+	c.mu.Lock()
+	// This happens if a user logs out when tailscale client is alreay logged
+	// out either because the user logged out previously or hasn't yet logged
+	// in.
+	if !c.loggedIn {
```

Comment:
Today, Logout seems like it also aborts an in-flight login attempt (one that was started but hasn't completed). Now it doesn't? Is that a problem?

Would it mean a user couldn't transition from the logging-in to to Stopped state in the state machine?

In any case, let's not land this right before the upcoming release per Denton's earlier announcement, lest it destabilize the various clients.

Reply:
> Today, Logout seems like it also aborts an in-flight login attempt (one that was started but hasn't completed). Now it doesn't? Is that a problem?

It doesn't seem to abort it at the moment.
If I run the cli login command from https://github.com/tailscale/tailscale/commit/c19b5bfbc391637b11c2acb3c725909a0046d849, wait for it to display the login URL, leave it hanging and run logout in parallel, the login remains hanging (I tried starting from both logged in and logged out state) and the logout results in the 'no nodekey to log out' error loop in both cases.
From this PR, the login remains hanging too.
In all cases the state ends up being `NeedsLogin` which as I understand it should be post-logout.
I might take a look at why the login is hanging and create a separate issue.

> In any case, let's not land this right before the upcoming release per Denton's earlier announcement, lest it destabilize the various clients.

👍🏼 I'll add a hold

Reply:
> Would it mean a user couldn't transition from the logging-in to to Stopped state in the state machine?

Thanks for the comment - that transition does not currently work, but seems like it would be more intuitive if that's how logout worked, so I've changed this in https://github.com/tailscale/tailscale/pull/8757